Created custom configurable/setting to ignore process names , (updated childProcessMonitor.ts and terminalConfiguration.ts)

san-coding opened the pull request:
<!-- Thank you for submitting a Pull Request. Please:
* Read our Pull Request guidelines:
  https://github.com/microsoft/vscode/wiki/How-to-Contribute#pull-requests
* Associate an issue with the Pull Request.
* Ensure that the code is up-to-date with the `main` branch.
* Include a description of the proposed changes and how to test them.
-->

This PR fixes #131386 

Added `zsh` to `ignoreProcessNames` array in `childProcessMonitor.ts` and created a new configurable/setting `terminal.integrated.confirmIgnoreProcesses` in  `terminalConfiguration.ts` , fixes  issue #131386 

### This PR resolves the issue that Background child processes created by `zsh` are not ignored by default , vscode already ignores some processes but a custom setting/configuration has been added to ignore processes.

- [x] I signed [CLA] [<img width="77" alt="Screenshot 2021-08-25 at 6 15 31 PM" src="https://user-images.githubusercontent.com/65719940/130792838-98f83a2a-f8b8-4e48-958b-054c28963402.png"> ](https://cla.opensource.microsoft.com/microsoft/vscode?pullRequest=131486l)
- [x] I reproduced the issue on my machine
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"

san-coding commented:
@Tyriar please review the pull request

Tyriar commented:
@san-coding thanks for the change, we've just started testing v1.60 so we'll probably review this for v1.61 next week

san-coding commented:
Thanks , will this be reviewed by the end of this month ?

san-coding commented:
@Tyriar is there a need to add a dropdown (enum) for the newly created configurable/setting ?

san-coding commented on src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts:
```diff
@@ -455,6 +455,11 @@ const terminalConfiguration: IConfigurationNode = {
 			type: 'boolean',
 			default: true
+		},
+		[TerminalSettingId.ignoreProcessNames]: {
```
Ok thanks, I will make those changes, but do I have the roll back my existing commits ?

san-coding commented:
@Tyriar I have changed `ignoreProcessNames` to `IgnoreProcessNames` in  `terminalConfiguration.ts`
 (Type rectified from last commit) and added `IgnoreProcessNames` definition to  `terminalPlatformConfiguration.ts`

Tyriar reviewed (changes requested):
This won't compile as `ignoreProcessNames` was removed but wasn't replaced with using the setting yet.

san-coding commented:
> This won't compile as `ignoreProcessNames` was removed but wasn't replaced with using the setting yet. Please make sure you test the change and that it fixed the issue.

Sorry I don't understand what you meant by _`ignoreProcessNames` was removed but wasn't replaced with using the setting yet_ , we are using `IConfigurationService` in place of it right ?

Tyriar commented:
> but I don't understand what you meant by ignoreProcessNames was removed but wasn't replaced with using the setting yet

You commented out the variable `ignoreProcessNames` but it's still being used and you didn't update the usage.

> we are using IConfigurationService in place of it right

You've injected the service but are not using it yet.

> could you please elaborate on why it won't compile

Please make sure the codebase compiles locally before sending PRs.

san-coding commented:
> > but I don't understand what you meant by ignoreProcessNames was removed but wasn't replaced with using the setting yet
> 
> You commented out the variable `ignoreProcessNames` but it's still being used and you didn't update the usage.
> 
> > we are using IConfigurationService in place of it right
> 
> You've injected the service but are not using it yet.
> 
> > could you please elaborate on why it won't compile
> 
> Please make sure the codebase compiles locally before sending PRs.

Yes I feel the below code will cause an error, how can I fix this 
https://github.com/microsoft/vscode/blob/ee1655a82ebdfd38bf8792088a6602c69f7bbd94/src/vs/platform/terminal/node/childProcessMonitor.ts#L123

san-coding commented:
> > > but I don't understand what you meant by ignoreProcessNames was removed but wasn't replaced with using the setting yet
> > 
> > 
> > You commented out the variable `ignoreProcessNames` but it's still being used and you didn't update the usage.
> > > we are using IConfigurationService in place of it right
> > 
> > 
> > You've injected the service but are not using it yet.
> > > could you please elaborate on why it won't compile
> > 
> > 
> > Please make sure the codebase compiles locally before sending PRs.
> 
> Yes I feel the below code will cause an error, how can I fix this , I don't see this being used anywhere else
> 
> https://github.com/microsoft/vscode/blob/ee1655a82ebdfd38bf8792088a6602c69f7bbd94/src/vs/platform/terminal/node/childProcessMonitor.ts#L123

san-coding commented:
## I have un-commented `ignoreProcessNames`, will the issue be resolved now ?

Tyriar commented:
> I have un-commented ignoreProcessNames, will the issue be resolved now ?

That will just do what it was doing before, you need to pull the setting in using the configuration service like this:

https://github.com/microsoft/vscode/blob/ee1655a82ebdfd38bf8792088a6602c69f7bbd94/src/vs/platform/terminal/node/terminalProfiles.ts#L42-L42

san-coding commented:
> > I have un-commented ignoreProcessNames, will the issue be resolved now ?
> 
> That will just do what it was doing before, you need to pull the setting in using the configuration service like this:
> 
> https://github.com/microsoft/vscode/blob/ee1655a82ebdfd38bf8792088a6602c69f7bbd94/src/vs/platform/terminal/node/terminalProfiles.ts#L42-L42

Ok I will try to fix the issue, should I pull setting in `childProcessMonitor.ts` ?

Tyriar commented:
Yep, in place of where ignoreProcessNames was being used

san-coding commented:
> Yep, in place of where ignoreProcessNames was being used

Hey @Tyriar , I made the changes, but I feel that  they are incorrect and syntax is wrong , could you please verify and tell me what to do, thanks

san-coding commented:
@Tyriar ??

Tyriar commented:
This is still not really that close. It's imperative that you actually test the change when you contribute to OSS before you submit the pull request so as to not waste the time of the maintainers. Also it's bad etiquette to ping people on twitter and email, github is the place for that. Closing this to prevent more time from being wasted on both our parts.

san-coding commented:
@Tyriar, I was just excited about my first pull request, wasn't receiving any response from your side, so I decided to ping you on Twitter and email you for help, I understand the inconvenience it might have caused, thank you for your assistance so far, I did learn a lot about VSCode codebase through this PR  and have learnt the lessons, won't repeat the same mistake, learning and contributing will continue 👍🏻

Tyriar commented:
@san-coding glad you learned something and I don't intend to me mean, but it's taking too much time for me to guide you through this. I also just saw you pinged me on unrelated PRs, this is really quite annoying. Here's a screenshot of my inbox to give you an idea of what I'm working with:

![image](https://user-images.githubusercontent.com/2193314/132673338-dcd1ed4f-e2ce-4219-80be-8573bdebeae0.png)

PRs are always welcome but they must work, this either did compile or didn't fix the problem it claimed to (using the setting).

san-coding commented:
> @san-coding glad you learned something and I don't intend to me mean, but it's taking too much time for me to guide you through this. I also just saw you pinged me on unrelated PRs, this is really quite annoying. Here's a screenshot of my inbox to give you an idea of what I'm working with:
> 
> ![image](https://user-images.githubusercontent.com/2193314/132673338-dcd1ed4f-e2ce-4219-80be-8573bdebeae0.png)
> 
> PRs are always welcome but they must work, this either didn't compile or didn't fix the problem it claimed to (using the setting).

Oh I understand .. will do better next time
